scipy/stats/_binned_statistic.py

```diff
@@ -584,7 +584,11 @@ def binned_statistic_dd(sample, values, statistic='mean',
             result[vv, a] = flatsum[a] / flatcount[a]
     elif statistic == 'std':
         result.fill(0)
-        _calc_binned_statistic(Vdim, binnumbers, result, values, np.std)
+        if values[0][0] != None:
```

Comment:
Can you add a comment about what is going on here? It's hard to understand from just reading the code - is this for `object` dtype?

Reply:
Yes, it is for the `object` type. I just thought it would be `object` type only when the input `values` is None. Seems it is better to do `if values.dtype == 'O':` ... I just modified this and added comments.